src/OrleansRuntime/Core/InsideRuntimeClient.cs

```diff
@@ -325,6 +326,12 @@ public void SniffIncomingMessage(Message message)
                         invokeExceptionLogger.Warn(ErrorCode.GrainInvokeException,
                             "Exception during Grain method call of message: " + message, exc1);
                     }
+
+                    if (exc1 is InconsistentStateException && target is Grain targetGrain)
```

Comment:
This is using a C# 7 feature. I don't think we made the jump to stop supporting vs 2015 yet (I think the idea is to do it right after 1.5)